Sources/SwiftProtobuf/Message+BinaryAdditions.swift

```diff
@@ -31,26 +31,24 @@ extension Message {
     if !partial && !isInitialized {
       throw BinaryEncodingError.missingRequiredFields
     }
-    let requiredSize = try serializedDataSize()
+    let requiredSize = try serializedBinarySize()
     var data = Data(count: requiredSize)
     try data.withUnsafeMutableBytes { (body: UnsafeMutableRawBufferPointer) in
-      if let baseAddress = body.baseAddress, body.count > 0 {
-        let pointer = baseAddress.assumingMemoryBound(to: UInt8.self)
+      let bytesWritten = try self.serializeBinary(into: body, partial: partial)
 
-        var visitor = BinaryEncodingVisitor(forWritingInto: pointer)
-        try traverse(visitor: &visitor)
-        // Currently not exposing this from the api because it really would be
-        // an internal error in the library and should never happen.
-        assert(requiredSize == visitor.encoder.distance(pointer: pointer))
-      }
+      // Currently not exposing this from the api because it really would be
+      // an internal error in the library and should never happen.
+      assert(requiredSize == bytesWritten)
     }
     return data
   }
 
   /// Returns the size in bytes required to encode the message in binary format.
-  /// This is used by `serializedData()` to precalculate the size of the buffer
-  /// so that encoding can proceed without bounds checks or reallocation.
-  internal func serializedDataSize() throws -> Int {
+  ///
+  /// This should be used with `serializeBytes(into:partial:)` so that enough space may
+  /// be allocated for the buffer so that encoding can proceed without bounds checks
+  /// or reallocation.
+  public func serializedBinarySize() throws -> Int {
     // Note: since this api is internal, it doesn't currently worry about
```

Comment:
I wasn't sure how to handle this comment: it seems like we shouldn't need to check for required fields as that's left to the serialize methods?

Reply:
Probably just promote it to a public comment.